src/main/java/org/skriptlang/skript/bukkit/recipes/elements/ExprAllRecipes.java

```diff
@@ -0,0 +1,197 @@
+			"[all [of]] [the] server['s] custom recipes [for %-itemstacks/itemtypes%]",
+			"[all [of]] [the] server['s] %recipetype% [for %-itemstacks/itemtypes%]",
+			"[all [of]] [the] server['s] (mc|minecraft|vanilla) %recipetype% [for %-itemstacks/itemtypes%]",
+			"[all [of]] [the] server['s] custom %recipetype% [for %-itemstacks/itemtypes%]");
+	}
+
+	private Expression<RecipeType> recipeTypeExpr;
+	private Expression<?> itemExpr;
+	private boolean getMinecraft, getCustom;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (matchedPattern <= 2 && exprs[0] != null) {
+			itemExpr = exprs[0];
+		} else if (matchedPattern >= 3) {
+			//noinspection unchecked
+			recipeTypeExpr = (Expression<RecipeType>) exprs[0];
+			itemExpr = exprs[1] != null ? exprs[1] : null;
+		}
+		getMinecraft = matchedPattern == 1 || matchedPattern == 4;
+		getCustom = matchedPattern == 2 || matchedPattern == 5;
+		return true;
+	}
+
+	@Override
+	protected Recipe @Nullable [] get(Event event) {
+		List<Recipe> recipes = new ArrayList<>();
+		getSelectedRecipes(event).forEachRemaining(recipes::add);
+		return recipes.toArray(new Recipe[0]);
+	}
+
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.RESET || mode == ChangeMode.DELETE)
+			return CollectionUtils.array(Recipe.class);
+		else if (mode == ChangeMode.ADD)
+			return CollectionUtils.array(Recipe[].class);
+		return null;
+	}
+
+	@Override
+	public void change(Event event, Object @Nullable [] delta, ChangeMode mode) {
+		switch (mode) {
+			case RESET -> {
+				Bukkit.resetRecipes();
+			}
+			case DELETE -> {
+				getSelectedRecipes(event).forEachRemaining(recipe -> {
+					if (recipe instanceof Keyed key)
+						Bukkit.removeRecipe(key.getKey());
+				});
+			}
+			case ADD -> {
+				Recipe[] recipes = delta != null ? (Recipe[]) delta : null;
+				if (recipes == null || recipes.length == 0)
+					return;
+				for (Recipe recipe : recipes) {
+					if (!(recipe instanceof Keyed keyed))
+						continue;
+					NamespacedKey key = keyed.getKey();
+					if (Bukkit.getRecipe(key) != null)
+						Bukkit.removeRecipe(key);
+					Bukkit.addRecipe(recipe);
+				}
+
+			}
+		}
+	}
+
+	@Override
+	public boolean isSingle() {
+		return false;
+	}
+
+	@Override
+	public Class<Recipe> getReturnType() {
+		return Recipe.class;
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		SyntaxStringBuilder builder = new SyntaxStringBuilder(event, debug);
+		builder.append("all of the");
+		if (getMinecraft) builder.append("minecraft");
+		else if (getCustom) builder.append("custom");
+		builder.append("recipes");
+		if (recipeTypeExpr != null) builder.append("of type", recipeTypeExpr);
+		if (itemExpr != null) builder.append("for", itemExpr);
+		return builder.toString();
+	}
+
+	private CheckedIterator<Recipe> getSelectedRecipes(Event event) {
```

Comment:
a few comments explaining the code here would be appreciated, it gets rather dense